### DISCUSSION TITLE
Issue 'hpp-fcl' not found

### DISCUSSION THREAD

**Discussion by kumar-sanjeeev** [stack-of-tasks/pinocchio#2496] [Q&A]
## Bug description
Installed pinocchio for Linux as mentioned. While compiling the overview-simple example, `hpp-fcl.pc` not found.

```
g++ -std=c++11 overview-simple.cpp -o overview-simple $(pkg-config --cflags --libs pinocchio)  

Package hpp-fcl was not found in the pkg-config search path.
Perhaps you should add the directory containing `hpp-fcl.pc'
to the PKG_CONFIG_PATH environment variable
Package 'hpp-fcl', required by 'pinocchio', not found
overview-simple.cpp:3:10: fatal error: pinocchio/multibody/sample-models.hpp: No such file or directory
    3 | #include "pinocchio/multibody/sample-models.hpp"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```
### Code
```
#include <iostream>
 
#include "pinocchio/multibody/sample-models.hpp"
#include "pinocchio/algorithm/joint-configuration.hpp"
#include "pinocchio/algorithm/rnea.hpp"
 
int main()
{
  pinocchio::Model model;
  pinocchio::buildModels::manipulator(model);
  pinocchio::Data data(model);
 
  Eigen::VectorXd q = pinocchio::neutral(model);
  Eigen::VectorXd v = Eigen::VectorXd::Zero(model.nv);
  Eigen::VectorXd a = Eigen::VectorXd::Zero(model.nv);
 
  const Eigen::VectorXd & tau = pinocchio::rnea(model, data, q, v, a);
  std::cout << "tau = " << tau.transpose() << std::endl;
}
```

### System

- OS: Ubuntu 22

**Comment by nim65s** [comment]
Hi @kumar-sanjeeev,

We would recommand to use CMake for your projects. A minimal example that you can use as a template is available at https://github.com/stack-of-tasks/pinocchio-minimal.

If you still want to use your compiler directly, the error message is explicit enough: you need to ensure pkg-config can find your hpp-fcl installation.

**Reply by kumar-sanjeeev** [reply]
@nim65s 
Thank you for recommendation. I tried with CMake, it works perfectly. 

I looked for `hpp-fcl` pkg-config at this path `/opt/openrobots/lib/pkgconfig`. I can see the `coal.pc` but no `hpp-fcl.pc`. Did I miss something while installation ?

**Reply by nim65s** [reply]
When we renamed hpp-fcl to coal, we provided an opt-in backward compatibility for the `hpp-fcl` name in C++ headers, python modules, and CMake exports, but it looks like we forgot pkg-config.

**Reply by nim65s** [reply]
A simple workaround would be to install hpp-fcl v2.4.5 instead of coal v3.0.0.
But anyways, CMake exports are more powerful and more tested than pkg-config.

**Reply by kumar-sanjeeev** [reply]
@nim65s 

I understand, and thank you for explaining and suggesting a simple workaround. I will stick with the CMake suggestion.
